### PR TITLE
CRS model

### PR DESCRIPTION
**Pull request by dionyziz**
- Introduce a CRS construction
- Show correctness unconditionally
- Show binding in standard model assuming collision resistance
- Show that private key cannot be found - hinting at unspendability
- Show uncensorability assuming [q] ~ H(t)

### CODE REVIEW

File: algorithms/alg.p2pk.tex [hunk 1]
```diff
@@ -0,0 +1,16 @@
+\begin{figure}[t]
+\begin{algorithm}[H]
+    \caption{\label{alg.p2pk} The blockchain address P2PK algorithm,
+    parameterized by a signature scheme $S = (\Gen, \Sig, \Ver)$.}
+    \begin{algorithmic}[1]
+        \Function{$\GenAddr_S$}{$1^\kappa$}
+            \Let{(pk, sk)}{\Gen(1^\kappa)}
+            \State\Return{pk}
+        \EndFunction
+        \Function{$\SpendVerify_S$}{$m, \sigma, pkh$}
```
Comment (gtklocker) [inline algorithms/alg.p2pk.tex]: `pkh`?

File: algorithms/alg.p2pk.tex [hunk 2]
```diff
@@ -0,0 +1,16 @@
+\begin{figure}[t]
+\begin{algorithm}[H]
+    \caption{\label{alg.p2pk} The blockchain address P2PK algorithm,
+    parameterized by a signature scheme $S = (\Gen, \Sig, \Ver)$.}
+    \begin{algorithmic}[1]
+        \Function{$\GenAddr_S$}{$1^\kappa$}
+            \Let{(pk, sk)}{\Gen(1^\kappa)}
+            \State\Return{pk}
```
Comment (gtklocker) [inline algorithms/alg.p2pk.tex]: the `sk` along with the `pkh` is returned in the p2pkh version, why is the `sk` not included here?

File: standard.tex [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+The simplest blockchain address protocol is the Pay to Public Key (P2PK) protocol which, in contrast to P2PKH does not hash the public key to generate an address. Instead, the address is literally the public key and spending verification simply checks the validity of a signature. This protocol is illustrated in Algorithm~\ref{alg.p2pk}.
+
+\import{./}{algorithms/alg.p2pk.tex}
+
+Without the Random Oracle model, our construction must be tailored to the signature scheme in order to ensure uncensorability, as our addresses must look similar to public keys generated by the scheme. We describe a burn scheme which can work for (EC)DSA signatures, as used in most cryptocurrencies today. Our scheme is unconditionally correct and binding in the standard model. We provide evidence of uncensorability in the Common Random String model, assuming the DLOG problem is hard, a collision resistant hash function exists. Additionally, we provide evidence that our scheme is unspendable in the Common Random String model and that no generic unspendable construction is possible in the standard model.
```
Comment (gtklocker) [inline standard.tex]: > We provide evidence of uncensorability in the Common Random String model, assuming the DLOG problem is hard, a collision resistant hash function exists.

doesn't make much sense

File: standard.tex [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+As evidence towards unspendability, we now remark that it is difficult for an adversary to obtain the secret key corresponding to the public key $h g^{H(t)}$ needed to produce signatures. We therefore conjecture that our scheme is unspendable.
+
+\begin{lemma}[Logarithm ignorance]
+  If $h$ is a \emph{Common Random String} and assuming the DLOG problem is hard, no probabilistic polynomial-time adversary can produce $(t, z)$ such that $g^z = h g^{H(t)}$, except with negligible probability in $\kappa$.
+\end{lemma}
+\begin{proof}
+  Suppose $\mathcal{A}$ is a probabilistic polynomial-time adversary which produces $(t, z)$ with probability of success $p = \Pr[g^z = h g^{H(t)}]$.
+  We construct the adversary $\mathcal{A}^*$ which invokes $\mathcal{A}$
+  illustrated in Algorithm~\ref{alg.secret-key-adversary} and finds the
```
Comment (gtklocker) [inline standard.tex:36]: alg.secret-key-adversary is not included

File: standard.tex [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+language. Additionally, they argue that such constructions are not possible in
+the standard model given non-uniform probabilistic polynomial-time adversaries,
+supporting our construction in the Common Random String model. Whether burn
+constructions in the Standard Model exist against uniform probabilistic
+polynomial-time adversaries remains a question for future work.
+
+\import{./}{algorithms/alg.collision-adversary-crs.tex}
+
+\begin{theorem}[Binding]
+  If the hash function $H$ is collision resistant and its range lies in $[q]$ where $q$ denotes the group order of $\mathbb{G}$, then the Proof of Burn protocol $\Pi$ of Algorithm~\ref{alg.construction-crs} is \emph{binding}.
```
Comment (gtklocker) [inline standard.tex]: shouldn't this be proof-of-burn?

File: standard.tex [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+  We thus conclude that $\mathcal{A^*}$ is successful in the $\collisionattack$ game if and only if $\mathcal{A}$ is successful in the $\bindattack$ game.
+
+  \[
+    \Pr[\bindattack_{\mathcal{A},\Pi} = \true]
+    =
+    \Pr[\collisionattack_{\mathcal{A}^*,H} = \true]
+  \]
+
+  From the collision resistance of $H$ it follows that $\Pr[\textsf{Collision}_{\mathcal{A}^*,H} = 1] < \negl$. Therefore,
```
Comment (gtklocker) [inline standard.tex]: Use \collisionattack and make the macro consistent with the other attack macros?

Comment (gtklocker) [reply standard.tex]: one line above it was \true instead of 1

File: standard.tex [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+  $\Pr[\textsf{Bind}_{\mathcal{A},\Pi} = 1] < \negl$, so
+  the protocol $\Pi$ is binding.
+\end{proof}
+
+We now give some evidence towards the uncensorability of our scheme.
+We introduce a helpful lemma on the nature of unpredictable
+distributions which shows that the converse of Lemma~\ref{lem:negl-pred}
```
Comment (gtklocker) [inline standard.tex]: you renamed this lemma to `lem:negl-unpred`

File: standard.tex [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+Initially, a $\kappa$-order multiplicative group $\mathbb{G}$ of order $q$ and a generator $g$ are selected and let the Common Random String be a random group element $h = g^y$ for some $y \in [q]$. Due to the self-reducibility of the DLOG problem, if DLOG is difficult in the group, an adversary will not be able to find the logarithm $y$ of the random group element, except with negligible probability.
+
+Our scheme is illustrated in Algorithm~\ref{alg.construction-crs}. $\GenBurnAddr$ hashes the tag $t$ and treats $H(t)$ as the exponent, calculates the public key $g^{H(t)}$ and blinds it using the factor $h$. As before, $\BurnVerify$ regenerates the burn address from $t$ and ensures it has been calculated correctly.
+
+\import{./}{algorithms/alg.construction-crs.tex}
+
+Correctness holds unconditionally.
+
+\begin{theorem}[Correctness]
+  The Proof of Burn protocol $\Pi$ of Algorithm~\ref{alg.construction-crs} is \emph{correct}.
```
Comment (gtklocker) [inline standard.tex]: capitalization

File: standard.tex [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+  between $X'$ and $Y'$. Then consider the probabilistic polynomial-time distinguisher $\mathcal{A}^*$ between $X$ and $Y$ illustrated in Algorithm~\ref{alg.pres-distinguisher}. Then
+  $\Pr[\distattack_{\mathcal{A}^*}(\kappa) = \true] =
+   \Pr[\distattack_{\mathcal{A}}(\kappa) = \true]$.
+  As $\Pr[\distattack_{\mathcal{A}^*}(\kappa) = \true] \leq \frac{1}{2} + \negl$, therefore $\Pr[\distattack_{\mathcal{A}}(\kappa) = \true] \leq \frac{1}{2} + \negl$.
+\end{proof}
+
+We are now ready to prove uncensorability.
+
+\begin{theorem}[Uncensorability]
+  Let $\mathcal{T}$ be an efficiently samplable unpredictable tag distribution and $H$ be a hash function such that $\{t \gets \mathcal{T}_\kappa; H(t)\} \cind \uniform([q])$ where $q$ denotes the order of the group $\mathbb{G}$. Then the Proof of Burn protocol $\Pi$ of Algorithm~\ref{alg.construction-crs} is \emph{uncensorable} with respect to blockchain address protocol $\Pi_\alpha$ of Algorithm~\ref{alg.p2pk}.
```
Comment (gtklocker) [inline standard.tex]: capitalization

**Review by gtklocker — CHANGES_REQUESTED** [review]

File: algorithms/alg.construction-crs.tex [hunk 10]
```diff
@@ -0,0 +1,16 @@
+\begin{figure}[t]
+\begin{algorithm}[H]
+    \caption{\label{alg.construction-crs} Our Uncensorable proof-of-burn protocol for P2PK using a Common Random String $h$ representing
```
Comment (gtklocker) [inline algorithms/alg.construction-crs.tex]: uncensorable should be lowercase to be uniform with the rest of the paper

File: standard.tex [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+    &\geq
+    \max_{h^* \in [X_\kappa]}\Pr[\collstar] - \Pr[t_1 = t_2]\,.
+  \end{align*}
+  Since $\Pr[\collisionattack_{\mathcal{A}}(\kappa) = \true] \leq \negl$
+  and $\Pr[t_1 = t_2] \leq \negl$, therefore $\max_{h^* \in [X_\kappa]} \Pr[\collstar] \leq \negl$.
+  Because $H(t_1)$ and $H(t_2)$ are chosen independently,
+  \[
+  \max_{h^* \in [X_\kappa]}\Pr_{x \gets X_\kappa}[x = h^*] = \sqrt{\max_{h^* \in [X_\kappa]} \Pr[\collstar]} \leq \negl\,.
+  \]
+  Applying Lemma~\ref{lem:unpred-negl}, we deduce that the distribution ensemble $X_\kappa$ is unpredictable.
```
Comment (gtklocker) [inline standard.tex]: How does this application look like?

File: standard.tex [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+  If $h$ is a \emph{Common Random String} and assuming the DLOG problem is hard, no probabilistic polynomial-time adversary can produce $(t, z)$ such that $g^z = h g^{H(t)}$, except with negligible probability in $\kappa$.
+\end{lemma}
+\begin{proof}
+  Suppose $\mathcal{A}$ is a probabilistic polynomial-time adversary which produces $(t, z)$ with probability of success $p = \Pr[g^z = h g^{H(t)}]$.
+  We construct the adversary $\mathcal{A}^*$ which invokes $\mathcal{A}$
+  illustrated in Algorithm~\ref{alg.secret-key-adversary} and finds the
+  logarithm of $h$.
+  Conditioned on the event that $\mathcal{A}$ is successful,
+  we have that
+  $g^z = h g^{H(t)} \Rightarrow g^z = g^{y + H(t)} \Rightarrow y \equiv z - H(t) \Modulo{q}$, so $\mathcal{A}^*$ is successful.
```
Comment (gtklocker) [inline standard.tex:40]: you need to introduce y as the logarithm of $h$ before

File: algorithms/alg.secret-key-adversary.tex [hunk 13]
```diff
@@ -0,0 +1,14 @@
+\begin{figure}[t]
+\begin{algorithm}[H]
+    \caption{\label{alg.secret-key-adversary}
+    The random discrete log solver $\mathcal{A}^*$
+    which makes use of an adversary $\mathcal{A}$ which
+    recovers the spending key corresponding to $h g^{H(t)}$.}
+    \begin{algorithmic}[1]
+        \Function{$\mathcal{A}^*_{\mathcal{A},H}$}{$h$}
+            \Let{(t, z)}{\mathcal{A}^*(h)}
+            \State\Return{$y - H(t)$}
```
Comment (gtklocker) [inline algorithms/alg.secret-key-adversary.tex]: you mean $z - H(t)$?

File: standard.tex (hunk 12 above)
Comment (dionyziz) [reply standard.tex:40]: It is already introduced here:

> Initially, a $\kappa$-order multiplicative group $\mathbb{G}$ of order $q$ and a generator $g$ are selected and let the Common Random String be a random group element $h = g^y$ for some $y \in [q]$. Due to the self-reducibility of the DLOG problem, if DLOG is difficult in the group, an adversary will not be able to find the logarithm $y$ of the random group element, except with negligible probability.

**Review by gtklocker — APPROVED** [review]

**Comment by gtklocker** [comment]
Anything pending to merge this?

**Review by gtklocker — APPROVED** [review]

**Pull request merged**